source/common/upstream/cluster_manager_impl.cc

```diff
@@ -86,6 +86,17 @@ getOrigin(const Network::TransportSocketOptionsConstSharedPtr& options, HostCons
   return {{"https", sni, host->address()->ip()->port()}};
 }
 
+bool isAdsCluster(const envoy::config::bootstrap::v3::Bootstrap& bootstrap,
+                  absl::string_view cluster_name) {
+  if (bootstrap.dynamic_resources().has_ads_config()) {
+    const auto& ads_config_source = bootstrap.dynamic_resources().ads_config();
+    return (ads_config_source.grpc_services_size() > 0 &&
```

Comment:
Does this need to address google_grpc as well?
If not, please add a comment justifying it.